src/engine.rs

```diff
@@ -442,6 +442,11 @@ impl Engine {
         // Effectively a dry-run
         let checkpoint = session.sandbox().take_snapshot();
         let properties = self.properties.clone();
+
+        // For each property, we will only try to break it once. If we find an argument
+        // that makes it return false, we will move on to the next property
```

Comment:
Hmm I think we will run up to max_rounds on each property unless we found a way to break it.
So if the property passes it means we tried max_rounds attempts with diffferents arguments and could not make the property return false (or traap?) 
If it Fails it means we found a way to make it return false (or trap) before we reached the maximun number of attempts...